actors/builtin/miner/miner_actor.go

```diff
@@ -656,6 +681,10 @@ func (a Actor) terminateSectors(rt Runtime, sectorNos []abi.SectorNumber, termin
 			if err != nil {
 				rt.Abort(exitcode.ErrIllegalState, "failed to delete sector: %v", err)
 			}
+
+			if !st.inChallengeWindow(rt) {
+				st.ProvingSet = st.Sectors
+			}
```

Comment:
In general I don't love that we have to do this on all sectors updates... But way more efficient than system that existed for surprise (which was necessary given the surprising nature of the thing).